library/src/main/java/com/liulishuo/filedownloader/FileDownloadMessageStation.java

```diff
@@ -166,6 +179,9 @@ public boolean handleMessage(Message msg) {
         private void dispose(final ArrayList<IFileDownloadMessenger> disposingList) {
             // dispose Sub-package-size each time.
             for (IFileDownloadMessenger iFileDownloadMessenger : disposingList) {
+                if (handleBlockCompleteMessage(iFileDownloadMessenger)) {
```

Comment:
`blockCompleteMessage` handled on the `dispose` area is not make sense, another better way?  I think all message need filter with `requestEnqueue`, so why it handles on the `requestEnqueue` is not enough?

Reply:
Yes, `blockCompleteMessage` should be handled on `requestEnqueue` method:
```java
    void requestEnqueue(final IFileDownloadMessenger messenger,
                        @SuppressWarnings("SameParameterValue") boolean immediately) {

        if (messenger.handoverDirectly()) {
            messenger.handoverMessage();
            return;
        }

        if (messenger.isBlockingCompleted()) {
            blockCompletedPool.execute(new Runnable() {
                @Override
                public void run() {
                    messenger.handoverMessage();
                }
            });
            return;
        }

        if (!isIntervalValid()) {
            ....
        }

        if (!isIntervalValid() || immediately) {
            ....
        }

        // enqueue.
        enqueue(messenger);
    }
```
But we can see, it depends on that `messenger.isBlockingCompleted()` return true, however, if `FileDownloadMessenger#parcelQueue` has unhandled progress message(it's possible if the consumer is slower than producer.),  `messenger.isBlockingCompleted()` will return false and the `blockCompleteMessage` will be added to `waitingQueue` and be handled on the `dispose` method at later.